### ISSUE TITLE
Race condition in startup

### ISSUE THREAD

**Issue by KnicKnic** [yyyar/gobetween#254]
I found this while trying to embed gobetween and not run it standalone so it may not hit as easily in normal case, but the race is still there.
https://github.com/yyyar/gobetween/blob/7d8a7368a535641df273c6b962750c1113d0c6b0/main.go#L83-L89
These calls all initialize defaults, however it is possible for the them not to complete before API calls get in. This then ends up in dereference of null.

I hit it in this case.
https://github.com/yyyar/gobetween/blob/7d8a7368a535641df273c6b962750c1113d0c6b0/src/manager/manager.go#L558

Ideally what main should do is run all of these synchronously to init them, and only then launch agents.

```go
api.Initialize()
metrics.Initialize()
manager.Initialize()
go api.Start()
go metrics.Start()
go manager.Start()
```

**Comment by yyyar** [comment]
Could you please show the exact error message you got and steps to reproduce?

**Comment by KnicKnic** [comment]
```
.\test.exe  
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0xa6c23e]

goroutine 1 [running]:
github.com/yyyar/gobetween/manager.prepareConfig(0xc129cb, 0x3, 0xc000180088, 0x0, 0x0, 0x0, 0xc1961f, 0xd, 0xc12abe, 0x3, ...)
        C:/Users/nmaliwa/go/src/github.com/yyyar/gobetween/src/manager/manager.go:572 +0xbce
github.com/yyyar/gobetween/manager.Create(0xc129cb, 0x3, 0x0, 0x0, 0x0, 0x0, 0xc1961f, 0xd, 0xc12abe, 0x3, ...)
        C:/Users/nmaliwa/go/src/github.com/yyyar/gobetween/src/manager/manager.go:193 +0x218
main.main()
        C:/Users/nmaliwa/go/src/github.com/yyyar/gobetween/src/test/do.go:26 +0x1e4
```
so I effectively coppied your main.go into launch\launch.go
I then used the launch package to embed into a test app and add servers using the manager package.

see my repro commit
https://github.com/KnicKnic/gobetween/commit/d454094dbc23010ed3abe40fcaaa0dd656377d98

**Comment by yyyar** [comment]
Thanks, I See. For standalone gobetween just reordering like this may be enough to fix a race condition:
```golang
 // Init metrics
metrics.Start((*cfg).Metrics)

// Init manager
manager.Initialize(*cfg)

// Start API
go api.Start((*cfg).Api)

```

**Comment by yyyar** [comment]
fixed by #261
